app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Any, Callable, Dict, Tuple
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache: Dict[Tuple, Any] = {}
+
+    def wrapper(*args) -> Any:
+        if args in func_cache:
```

Comment:
You should check both `args` and `kwargs` because the function might accept the same `args` but different `kwargs`. Additionally, I recommend checking the function name since there could be two different functions that accept the same arguments.